lib/driver.js

```diff
@@ -412,10 +412,20 @@ class XCUITestDriver extends BaseDriver {
       await this.wda.uninstall();
       this.logEvent('wdaUninstalled');
     } else if (!util.hasValue(this.wda.webDriverAgentUrl) && (await this.wda.isRunning())) {
-      log.info(`Will reuse previously cached WDA instance at '${this.wda.url.href}'. ` +
-               `Set the wdaLocalPort capability to a value different from ${this.wda.url.port} ` +
-               `if this is an undesired behavior.`);
-      this.wda.webDriverAgentUrl = this.wda.url.href;
+      let currentAppBundleID;
+      if (util.hasValue(this.opts.updatedWDABundleId)) {
+        let response = await this.proxyCommand('/status', 'GET');
```

Comment:
const response

Reply:
I don't think this will work, since `proxyCommand` without the `isSessionCommand` argument set to false will use the normal proxy. It's also entirely possible that we don't have a proxy yet, so this will error out.

Reply:
yeah, there is a separate NoSessionProxy wrapper for this purpose

Reply:
Yes. `response = await this.proxyCommand('/status', 'GET', null, false);` ought to use it. Except those proxies are only initialized when `wda.launch` is called.